pennylane/templates/layers/gate_fabric.py

```diff
@@ -0,0 +1,235 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the QuantumNumberPreserving Gate Fabric template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+import numpy as np
+import pennylane as qml
+from pennylane.operation import Operation, AnyWires
+
+
+class GateFabric(Operation):
+    r"""Implements a local, expressive, and quantum-number-preserving ansatz using VQE circuit fabrics
+    proposed by Anselmetti *et al.* in `arXiv:2104.05692 <https://arxiv.org/abs/2104.05695>`_.
+
+    This template prepares the :math:`N` qubits trial state by applying :math:`L` layers of the gate fabric, which is composed
+    of 2-parameter 4-qubit gate elements :math:`\hat{Q}(\theta, \varphi)`. Each of the :math:`\hat{Q}(\theta, \varphi)` gate is itself
+    composed of two 1-parameter 4-qubit gates. The first gate is a spin-adapted spatial orbital rotation gate, which is implemented by
+    :math:`\text{OrbitalRotation}(\varphi)` gate. Whereas the second gate is a diagonal pair-exchange gate, which is implemented by the
```

Comment:
Please use cross referencing for `OrbitalRotation` and `DoubleExcitation` as explained [here](https://pennylane.readthedocs.io/en/stable/development/guide/documentation.html#guiding-principles).

Reply:
Done.